sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -123,6 +123,31 @@ class SubquerySuite extends QueryTest with SharedSQLContext {
     )
   }
 
+  test("SPARK-15677: Scalar sub-query in Select list against a DataFrame generated query") {
+    Seq((1, 1), (2, 2)).toDF("c1", "c2").createOrReplaceTempView("t1")
```

Comment:
please use `withTempTable`(ok the name is wrong for history reasons, it should be `withTempView`), which will drop the view after the test for you.